src/main/java/seedu/address/model/ingredient/IngredientQuantity.java

```diff
@@ -134,7 +136,7 @@ public IngredientQuantity subtract(IngredientQuantity other) {
 
         assert newValue != null;
         if (newValue.doubleValue() < 0) {
-            newValue = value;
+            newValue = 0;
```

Comment:
I've also changed this to throw a `NonPositiveIngredientQuantityException` instead, and shift the handling of the exception to the `UniqueIngredientList`/`CompatibleIngredientList`, where the current implementation is to simply remove these items from the list entirely. In future, we could consider if we want to push the exception to the user with the intention of letting them correct a typo.